src/bp/ui-admin/src/Pages/Account/Login.tsx

```diff
@@ -1,208 +1,113 @@
-import { Button, FormGroup, InputGroup, Intent } from '@blueprintjs/core'
+import { AuthStrategyConfig } from 'common/typings'
 import { get } from 'lodash'
-import React, { Component, Fragment } from 'react'
+import React, { FC, useEffect, useState } from 'react'
 import { RouteComponentProps } from 'react-router'
 import { Redirect } from 'react-router-dom'
-import { Alert, Card, CardBody, CardText, CardTitle } from 'reactstrap'
 
 import api from '../../api'
-import logo from '../../media/nobg_white.png'
-import { setActiveWorkspace } from '../../Auth'
+import BasicAuthentication, { setActiveWorkspace } from '../../Auth'
+import { LoginContainer } from '../Layouts/LoginContainer'
 
-interface Params {
-  strategy: string
-  workspace: string
-}
+import { AuthMethodPicker } from './AuthMethodPicker'
+import { LoginForm } from './LoginForm'
 
 type Props = {
-  auth: any
-} & RouteComponentProps<Params>
-
-interface State {
-  isLoading: boolean
-  isFirstTimeUse: boolean
-  email: string
-  strategyType: string
-  authEndpoint: string
-  password: string
-  passwordExpired: boolean
-  error?: string
-  loginUrl: string
-  strategyId: string
+  auth: BasicAuthentication
+} & RouteComponentProps<{ strategy: string; workspace: string }>
+
+const extractErrorMessage = () => {
+  const urlParams = new window.URLSearchParams(window.location.search)
+  return urlParams.get('error')
 }
 
-export default class Login extends Component<Props, State> {
-  state: State = {
-    isLoading: true,
-    isFirstTimeUse: false,
-    passwordExpired: false,
-    strategyType: 'basic',
-    strategyId: '',
-    authEndpoint: '',
-    email: '',
-    password: '',
-    loginUrl: '',
-    error: undefined
-  }
+const Login: FC<Props> = props => {
+  const [isLoading, setLoading] = useState(true)
+  const [isFirstUser, setFirstUser] = useState(false)
+  const [strategies, setStrategies] = useState<AuthStrategyConfig[]>()
+  const [loginUrl, setLoginUrl] = useState('')
+  const [error, setError] = useState<string | null>()
 
-  loadAuthConfig = async () => {
-    const { isFirstUser, ...strategyConfig } = await this.props.auth.getStrategyConfig(this.props.match.params.strategy)
+  useEffect(() => {
+    setError(undefined)
```

Comment:
why don't you set the default value at line 28 when using useState hook ?

Reply:
The error message is reset everytime the strategy is changed (ex: by updating the URL)